pdg/src/main.rs

```diff
@@ -50,10 +51,104 @@ impl Display for ToPrint {
     }
 }
 
+pub struct Pdg {
+    pub events: Vec<Event>,
+    pub metadata: Metadata,
+    pub graphs: Graphs,
+}
+
+impl Pdg {
+    pub fn new(metadata_path: &Path, event_log_path: &Path) -> eyre::Result<Self> {
+        let events = read_event_log(event_log_path)?;
+        let metadata = read_metadata(metadata_path)?;
+        let graphs = construct_pdg(&events, &metadata);
+        Ok(Self {
+            events,
+            metadata,
+            graphs,
+        })
+    }
+
+    pub fn repr<'a>(&'a self, to_print: &'a [ToPrint]) -> PdgRepr<'a> {
```

Comment:
I think `PdgRepr` should be constructed from a `Pdg`, not the other way around

Reply:
i.e. `PdgRepr::new` should do this IMO

Reply:
Can `PdgRepr::new` be the real implementation and then still have a `Pdg::repr` method?

Reply:
Actually, the fields of `PdgRepr` are `pub`, so it can be constructed on its own.  A `PdgRepr::new` wouldn't really offer any advantage, and I'd like to keep `Pdg::repr` for ergonomics.

Reply:
I'm not sold on the utility of `Pdg` knowing anything about what a `PdgRepr` is, that's my main concern. I think it should be constructed on its own, how is less important.

Reply:
I see `PdgRepr` more as an `impl Display` for `Pdg`, but with `&[ToPrint]`s, too.  I'm not sure why `Pdg` can't know about `PdgRepr`, especially as its such a simple and transparent struct.

Reply:
I'm going to merge this as is if that's okay.  We can easily change it back afterward if we need to as it's only a couple of lines of code.